### PR TITLE
[M1346] Add table and promise caching

### PR DESCRIPTION
**Pull request by gridcell**

### CODE REVIEW

**Comment by gridcell** [comment]
[sites listing still slow after editing](https://trello.com/c/MMSMhyma/1346-sites-listing-still-slow-after-editing)

File: src/app.config.js [hunk 1]
```diff
@@ -363,7 +363,7 @@ appConfig.httpErrorWhitelist = [
   { status: 403, detail: 'Forbidden.' },
   { status: 404, detail: 'Not found.' }
 ];
-appConfig.defaultCacheExpire = 1; // in minutes
+appConfig.defaultCacheExpire = 60000; // in minutes
```
Comment (saanobhaai) [inline src/app.config.js:366]: Is this really what we want for the default? Or have the units changed?

File: src/app/_mermaid_common/libs/services/cache.svc.js [hunk 2]
```diff
@@ -1,32 +1,61 @@
...
-      expire = expire || this.expire;
-      this._cache.set(key, value, expire);
-    },
-    get: function(key) {
-      var v = this._cache.get(key);
-      if (v !== null) {
-        console.log('"%s" from cache', key);
-      }
-      return v;
+
```
Comment (saanobhaai) [inline src/app/_mermaid_common/libs/services/cache.svc.js:3]: This is awesome, but are we sure that nothing else in the existing code uses things like `removeAll` or `setBucket`?

Comment (gridcell) [reply src/app/_mermaid_common/libs/services/cache.svc.js:3]: yeah I did a search

File: src/app.config.js (hunk 1 above)
Comment (gridcell) [reply src/app.config.js:366]: I'll update comment, it's in `ms` now

**Pull request merged**